Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -241,9 +317,17 @@ fileprivate extension Compiler.ByteCodeGen {
           ? input.index(after: bounds.lowerBound)
           : nil
       }
-    } else {
-      builder.buildMatch(c)
     }
+    
+    if optimizationsEnabled && c.isASCII {
+      for scalar in c.unicodeScalars {
+        let boundaryCheck = scalar == c.unicodeScalars.last!
```

Comment:
I think you need to check indices. You could have the same combining acute accent appear multiple times. 

Actually, could you write a test of exactly that while we have the bug here?

Reply:
If we have a combining accent then c would not be ascii so I don't think the old code has a bug, the only multi scalar ascii character is CR-LF. In any case index checking is cleaner and doesn't have to worry about edge cases like that so I'll rewrite it to do that